src/include/sof/drivers/mu.h

```diff
@@ -72,12 +72,12 @@ enum imx_mu_xsr {
 
 static inline uint32_t imx_mu_read(uint32_t reg)
 {
-	return *((volatile uint32_t*)(MU_BASE + reg));
+	return *((volatile uint32_t*)((uintptr_t)(MU_BASE + reg)));
 }
 
 static inline void imx_mu_write(uint32_t val, uint32_t reg)
 {
-	*((volatile uint32_t*)(MU_BASE + reg)) = val;
+	*((volatile uint32_t*)((uintptr_t)(MU_BASE + reg))) = val;
```

Comment:
Again, MU_BASE is a literal so it is really not intuitive why this is needed. Please add a short explanation in the code (and this one does not seem explained by the commit message)

Reply:
Please do this instead (tested)

```diff
--- a/src/platform/imx93_a55/include/platform/lib/memory.h
+++ b/src/platform/imx93_a55/include/platform/lib/memory.h
@@ -49,7 +49,7 @@
 #define HEAPMEM_SIZE 0x00010000
 
 /* SOF uses A side of the WAKEUPMIX MU */
-#define MU_BASE 0x42430000
+#define MU_BASE ((uintptr_t)0x42430000)
 
 /* SOF uses EDMA2 (a.k.a EDMA4 in the TRM) */
 #define EDMA2_BASE 0x42010000

```

Fun fact: C++23 allows `0x42430000Z`

Reply:
Fixed this by adding the `ULL` suffix. This ought to make `MU_BASE` 64-bit.